subsys/net/lib/lwm2m/lwm2m_obj_device.c

```diff
@@ -185,6 +185,12 @@ int lwm2m_device_add_err(uint8_t error_code)
 		if (error_code_ri[i].res_inst_id == RES_INSTANCE_NOT_CREATED) {
 			break;
 		}
+
+		/* No duplicate error codes allowed */
+		if (memcmp(error_code_ri[i].data_ptr, &error_code,
```

Comment:
As the `error_code` is a byte, perhaps something like `*(uint8_t *)error_code_ri[i].data_ptr == error_code` would be better here?

Reply:
sure

Reply:
kk @jukkar